include/ensmallen_bits/sgd/sgd_impl.hpp

```diff
@@ -138,9 +143,16 @@ double SGD<UpdatePolicyType, DecayPolicyType>::Optimize(
   overallObjective = 0;
   for (size_t i = 0; i < numFunctions; i += batchSize)
   {
-    const size_t effectiveBatchSize = std::min(batchSize, numFunctions - i);
-    overallObjective += f.Evaluate(iterate, i, effectiveBatchSize);
+    size_t effectiveBatchSize = std::min(batchSize, numFunctions - i);
```

Comment:
First we calculate effectiveBatchSize and if it is 0 then make effectiveBatchSize to 1 for calculating overallObjective and increment increment i by one. Then call evaluate with iterate, i-1( as we incremented it before passing ) and effectiveBatchSize. 

++i is done to move the loop forward because batchSize is 0 which means i+=batchSize doesn't update i and we will be trapped in infinite looping.